Source/Core/Common/BitField.h

```diff
@@ -124,15 +124,17 @@ struct BitField
   // so that we can use this within unions
   constexpr BitField() = default;
 
+// Visual Studio (as of VS2017) considers BitField to not be trivially
+// copyable if we delete this copy assignment operator.
+#ifndef _MSC_VER
```

Comment:
I really don't think applying this on a platform-specific basis is a really good idea. This is just as much a 'hack' around the problem itself. You might want to ensure we're even following the requirements for [IsTriviallyCopyable](http://en.cppreference.com/w/cpp/concept/TriviallyCopyable) in the first place.

Reply:
> I really don't think applying this on a platform-specific basis is a really good idea. This is just as much a 'hack' around the problem itself.

Yeah... This PR exchanges one ifdef hack for another :/

> You might want to ensure we're even following the requirements for IsTriviallyCopyable in the first place.

We are following it as far as I can see – we have no non-trivial copy/move constructors or copy/move assignment operators, the destructor is trivial, and we have an auto-generated copy constructor. (Or at least I think the copy constructor gets generated – I don't remember the exact rules for what happens when we have a deleted copy assignment operator – but explicitly marking the copy constructor as `default` didn't change anything in my tests with VS, so this part doesn't seem to be the problem.) So it seems likely that the class is trivially copyable according to the requirements, considering that all compilers other than VS say so, but I'm not an expert in this area.

Reply:
I looked into reporting this as a bug now, but it seems like it already has been reported: https://developercommunity.visualstudio.com/content/problem/101208/c-compiler-is-overly-strict-regarding-whether-a-cl.html

So hopefully this will get fixed in a future version, but for now, I think having a workaround for this bug at least isn't any worse than stubbing out IsTriviallyCopyable.

Reply:
Yeah, if it's a bug in the compiler, there's not much we can do, so it's fine. However, it's probably be beneficial to include the link to the issue, or put some means of finding the issue in a comment, so it can be re-examined later.

Reply:
I've added that link to the comment.